thedmstoolkit/character_generation.py

```diff
@@ -0,0 +1,538 @@
+
+        Raises:
+            RuntimeError: If the key provided is the wrong type
+            RuntimeError: If there is an unauthorized item in the passed list
+
+        Returns:
+            String: Alignment
+        """
+        if GeneratorKey not in Character_Generator.Generators:
+            raise RuntimeError("Generator does not exist")
+        for x in alignment_list:
+            if x not in Character_Generator.ALIGNMENT_DICT["All"]:
+                raise RuntimeError("Invalid List")
+        alignment = alignment_list[
+            Character_Generator.Generators[GeneratorKey](0, len(alignment_list) - 1)
+        ]
+        return alignment
+
+    @staticmethod
+    def generate_background(background_list: list[str], GeneratorKey: str) -> str:
+        """
+            Generate a random background
+        Args:
+            background_list (str): background
+            GeneratorKey (str): key required to specify which randomizer you want
+
+        Raises:
+            RuntimeError: If the key provided is the wrong type
+            RuntimeError: If there is an unauthorized item in the passed list
+
+        Returns:
+            String: Background
+        """
+        if GeneratorKey not in Character_Generator.Generators:
+            raise RuntimeError("Generator does not exist")
+        for x in background_list:
+            if x not in Character_Generator.BACKGROUND_LIST:
+                raise RuntimeError("Invalid List")
+        return background_list[
+            Character_Generator.Generators[GeneratorKey](0, len(background_list) - 1)
+        ]
+
+    @staticmethod
+    def Generate(
+        generations_list: Optional[list[str]] = None,
+        stat_generator_key: Optional[str] = None,
+        race_key="All",
+        class_key="All",
+        alignment_key="All",
+        GeneratorKey="random",
+        stat_list=Optional[list[int]],
+    ) -> dict[str, Union[list[int], str]]:
+        """Given generator parameters, return a dictionary of character characteristics
+
+        Args:
+            generations_list (str, optional): A list of the various generations to do. Defaults to "All".
+            stat_generator_key (Optional[str], optional): A generator name from limited_generators(). Defaults to None.
+            race_key (str, optional): A race category, from the RACE_DICT. Defaults to "All".
+            class_key (str, optional): A class category from the CLASS_DICT. Defaults to "All".
+            alignment_key (str, optional): A alignment category from the ALIGNMENT_DICT. Defaults to "All".
+            stat_list (list[int], optional): A list of 6 numbers between 1 and 20 to use as stats. Defaults to None.
+
+        Raises:
+            ValueError: _description_
+            RuntimeError: Not all numbers are in the passed stat array are valid
+
+        Returns:
+            dict[str, list[int] | str]: Generated Characteristics
+        """
+
+        if type(stat_list) is list:
```

Comment:
maybe more useful to use a "guard clause" here
```python
if type(stat_list) not list:
  raise ValueError()
```

Reply:
Except we are fine if it's not a list in the case it's not entered so that's not an option